### DISCUSSION TITLE
Support more use cases for directus-extension CLI

### DISCUSSION THREAD

**Discussion by t7tran** [directus/directus#6639] [Feature Requests]
The current implementation of `directus-extension` CLI seems to support a simple use case as presented in the Docs' examples. In reality, we tend to have multiple extension types to be developed and deployed in a single Directus instance. While building the support for more use cases into the CLI tool proves to be challenging, it will ease the learning curve for the beginners, as well as provide the starting point for those who want to extend it further.

Please comment on features you would like the CLI to support:

- [ ] Mutilple input
- [ ] Multiple corresponding output
- [ ] Custom plugins (e.g. `YAML`, `typescript`) and their orders

Below is our current `rollup.config.js` for your reference.

```
import { terser } from "rollup-plugin-terser";
import resolve from "rollup-plugin-node-resolve";
import commonjs from "rollup-plugin-commonjs";
import vue from "rollup-plugin-vue";
import multiInput from "rollup-plugin-multi-input";
import typescript from "rollup-plugin-typescript";
import styles from "rollup-plugin-styles";
import replace from "@rollup/plugin-replace";
import yaml from "@rollup/plugin-yaml";
import { SHARED_DEPS } from '@directus/shared/constants';

function watchChange() {
  return {
    name: "watch-change",
    watchChange(id) {
      // signal to restart Directus on build changes
    }
  };
}

export default {
  input: [
    "src/displays/*/index.ts",
    "src/interfaces/*/index.ts",
    "src/layouts/*/index.ts",
    "src/modules/*/index.ts"
  ],
  output: {
    format: "es",
    dir: "rootfs/directus/extensions",
  },
  external: SHARED_DEPS,
  plugins: [
    multiInput(),
    replace({
      "process.env.NODE_ENV": JSON.stringify("production"),
      preventAssignment: true
    }),
    resolve({
      mainFields: ["browser", "module", "main"]
    }),
    vue(),
    styles(),
    yaml(),
    typescript(),
    commonjs(),
    terser(),
    watchChange()
  ]
};
```

**Comment by rijkvanzanten** [comment]
cc @nickrum

**Reply by t7tran** [reply]
This discussion is similar to #6599 as @nickrum pointed out on Discord. Could someone merge them up?

**Reply by t7tran** [reply]
Another line of thoughts, would it be even more practical to expose an official monorepo including a showcase of all extension types? One just needs to clone; optionally modifiy the relevant parts; optionally comment out the non-relevant part then they can see everything in action. It can be used as a playground or a starter.

The project will need to have proper releases with change log so people can update their existing code. We've received tremendous support from the team/community and are happy to contribute to this.

As it progresses, nice features can be cherry-picked into the main application, e.g. translations.

**Reply by rijkvanzanten** [reply]
> One just needs to clone; optionally modifiy the relevant parts; optionally comment out the non-relevant part then they can see everything in action. It can be used as a playground or a starter.

This is more or less what https://github.com/directus/directus/pull/6590 tries to achieve 🙂

**Reply by t7tran** [reply]
Even better. Then keep the original post to consider support those use cases consistently in both `create` and `build` command.

By the way, when are we going to make extensions' **transations** a first class citizen? With the upgrade work almost completed, we got it to work the same way as the main app. However, the loading mechanism is a bit repetitive and doesn't support translation outside of the component, i.e. the module index.ts's `name`.

**Reply by benhaynes** [reply]
I believe there will be a focus on extensions over the coming month or so. Formal translations in extensions (managed in the core) is something we're actively talking about.

**Comment by nickrum** [comment]
Alongside npm extensions we will support a `pack` extension type that serves as a way to group together extensions by listing them as dependencies. We could support bundling multiple extensions by bundling any dependent extension instead when calling `directus-extension build` inside the folder of a `pack` extension.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
